features/0056-service-decorator/README.md

```diff
@@ -19,6 +19,12 @@ Examples include the Connect Protocol and Challenge Protocols.
 
 The `~service` decorator on a message contains the service definition that you might expect to find in a DID Document. These values function the same way.
 
+The `~service` decorator may also be used for protocols which involve more than a simple request and response.  For example, using the `~service` decorator with the
+Present Proof Protocol allows a prover to present a proof to a verifier when the prover and verifier do not have a pre-existing connection or relationship.
+In this case, the `~service` decorator must be used on all messages of the Present Proof Protocol.
+
+More generally, when a `~service` decorator is used for any single message of a protocol, the `~service` decorator MUST be present on ALL messages of that protocol.
```

Comment:
I'm not sure this matters because I think the need for the ~service decorator becomes unnecessary in the request/response use case if the Out of Band PR/RFC happens.  However, I would argue that when used with the Present Proof protocol, it is a simple request/response flow. Further, I'm not sire the MUST on that last statement is true and definitely not enforceable. The agents can/will do what they want in using the ~service decorator.  We've yet to see the ~service decorator used on other than a request/response, so I'm not sure of a practical use case for this.

Reply:
Stephen, what "Out of Band PR/RFC" are you referring to?  Can you elaborate?  It would help me (and perhaps others) to see specific examples of how the service decorator is being used today.  And if it should definitely not be used in specific cases, this should also be stated in the RFC.  And if some or all of this info is stated elsewhere, please point me to it and I'll be happy to create a PR to this RFC which references the others.

When used with the Present Proof Protocol, I'm thinking of the following message flow:

1. Verifier sends presentation request to prover

2. Prover sends presentation to verifier

3. Verifier sends ACK to prover

This is more than a single request/response flow given the additional ACK.  And using the service decorator in this flow has use cases I believe.

For example, Alice approaches a Bartender who needs proof that Alice is 21.  There is no connection between Alice and Bartender, so Alice begins the verification flow with a presentation proposal message to Bartender with a service decorator.